### DISCUSSION TITLE
No Passthru Devices found on this System

### DISCUSSION THREAD

**Discussion by Hypfer** [rnd-ash/OpenVehicleDiag#28] [Q&A] [answered]
Hi,

so I just got my A0 and tried using this software. After various issues with no vulkan support, an outdated glibc etc, I can now run the software either on a W10 install or a fresh Ubuntu 21.04 install.

Sadly, it won't detect the A0. Neither using the official firmware nor using the custom firmware found here:
https://github.com/rnd-ash/Macchina-J2534

Unfortunately, I can't provide any logs since there are none.

Any pointers where to go from here?

**Comment by rnd-ash** [comment, marked as answer]
Thats odd, did you compile OpenVehicleDiag from scratch?
 Regarding the A0 driver, did you merge the .REG file?

**Reply by Hypfer** [reply]
Oh I'm just stupid, sorry.
I didn't realize that the driver is more than the firmware. So the problem was me not properly reading documentation.

Anyways good news: it works
Bad news: My car now thinks that it has a broken 4WD

![image](https://user-images.githubusercontent.com/974410/121081219-b0ed3400-c7dc-11eb-8bab-8d87ee494015.png)

![image](https://user-images.githubusercontent.com/974410/121081229-b480bb00-c7dc-11eb-8507-13d5540a0636.png)


Unfortunately, the error message doesn't seem to go away just by power cycling the car.
Fortunately it still drives so that's a relief

OVD did segfault before the scan finished so I don't have a report to share either. Attempts to rescan after this fault occoured led to a constant spam of `Rx Queue in Channel 0 is full` and nothing else happening.

While I try to figure out how to convince my car that it has actually only 2WD, I'd propose that OVD should persist the scan state to disk during scanning so that even a crash won't let all scan results vanish.

Btw it's a 2015 Nissan Pulsar C13.

I somewhat doubt that my dealership will believe me when I tell them that it spontaneously started identifying as a 4x4 though. We'll see

**Reply by Hypfer** [reply]
Interestingly, even removing the battery completely for a few minutes did not resolve this issue.

Considering that OVD never got to stage 3,  I guess it's safe to say that 
https://github.com/rnd-ash/OpenVehicleDiag/blob/98fc634448fa3504548c998e60a27a469b209b74/app_rust/src/windows/diag_scanner.rs#L225-L242

may not always be a safe message since it apparently did change the coding of some ECU, which is surprising to me considering that it's just an empty frame.

Maybe there's some kind of autoconfig going on with ECUs starting to assume that another ECU might exist based on them seeing CAN messages with e.g. the ID of the AWD ECU? Does stuff like that exist?

@rnd-ash Should I move this to an issue/another discussion?

**Reply by Hypfer** [reply]
> so I don't have a report to share either.

As it turns out, I do have some things to share. Namely the macchina_log.txt

[macchina_log.txt](https://github.com/rnd-ash/OpenVehicleDiag/files/6614669/macchina_log.txt)

It consists of a scan with the ignition off since I'm an idiot followed by some driving around to charge the battery followed by the 4WD-causing scan with the ignition on

While I'd definitely want to continue researching this, I'm also mildly scared that the next scan will mess up the car in a way that it stops driving

**Reply by Hypfer** [reply]
Update:

I was able to solve this by disconnecting the battery and keeping it disconnected for ~6h
That seems to have reset everything in the car.

Interestingly, the 4x4-i view also went away and I'm not able to enable it in the menu again so there's definitely some kind of auto-discovery of ECUs and features based on Message-IDs seen going on
